src/main/java/gov/cabinetoffice/gapfindapiadmin/controllers/ApiKeyController.java

```diff
@@ -61,9 +64,29 @@ public ModelAndView createKey(final @Valid @ModelAttribute CreateApiKeyDTO creat
                 .addObject("keyValue", apiGatewayService.createApiKeysInAwsAndDb(createApiKeyDTO.getKeyName()));
     }
 
-    //TODO change this to a better handling
+    @GetMapping("/revoke/{apiKeyId}")
+    public ModelAndView showRevokeApiKeyConfirmation(@PathVariable int apiKeyId) {
+        Optional<ApiKey> apiKey = apiKeyService.getApiKeyById(apiKeyId);
+
+        if(!apiKey.isPresent()) {
+            return new ModelAndView("redirect:/api-keys");
+        }
```

Comment:
We could put this on one line and do an elseThrow(), and then the exception handler could return a message to the api keys page saying "invalid api key" or something. One for future though when we look at better error handling as another ticket.